src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -129,6 +133,93 @@ void RaftReplDev::push_data_to_all_followers(repl_req_ptr_t rreq) {
         });
 }
 
+void RaftReplDev::on_fetch_data_received(intrusive< sisl::GenericRpcData >& rpc_data) {
+    auto const& incoming_buf = rpc_data->request_blob();
+    auto fetch_req = GetSizePrefixedFetchData(incoming_buf.cbytes());
+
+    RD_LOG(INFO, "Data Channel: FetchData received: fetch_req.size={}", fetch_req->request()->entries()->size());
+
+    std::vector< sisl::sg_list > sgs_vec;
+
+    struct Context {
+        std::condition_variable cv;
+        std::mutex mtx;
+        size_t outstanding_read_cnt;
+    };
+
+    auto ctx = std::make_shared< Context >();
+    ctx->outstanding_read_cnt = fetch_req->request()->entries()->size();
+
+    for (auto const& req : *(fetch_req->request()->entries())) {
+        RD_LOG(INFO, "Data Channel: FetchData received: lsn={}", req->lsn());
+
+        auto const& lsn = req->lsn();
+        auto const& term = req->raft_term();
+        auto const& dsn = req->dsn();
+        auto const& header = req->user_header();
+        auto const& key = req->user_key();
+        auto const& originator = req->blkid_originator();
+        auto const& remote_blkid = req->remote_blkid();
+
+        // fetch data based on the remote_blkid
+        if (originator == server_id()) {
+            // We are the originator of the blkid, read data locally;
+            MultiBlkId local_blkid;
+
+            // convert remote_blkid serialized data to local blkid
+            local_blkid.deserialize(sisl::blob{remote_blkid->Data(), remote_blkid->size()}, true /* copy */);
```

Comment:
They should be strictly same size, since this is a deserialize/serialize which provided by MultiBlkId itself. There is unit test case cover these apis in MultiBlkId.